src/features/importer/Uploader.js

```diff
@@ -0,0 +1,150 @@
+  }
+}
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  return {
+    setZoom: (event) => {
+      dispatch(updateImporter({zoom: parseFloat(event.target.value)}))
+    },
+    toggleAspectRatio: (event) => {
+      dispatch(toggleFileAspectRatio())
+    },
+    onChange: (attrs) => {
+      dispatch(updateImporter(attrs))
+    },
+  }
+}
+
+class Uploader extends Component {
+  onFileImported(importer, importerProps) {
+    this.props.onChange(importerProps)
+
+    this.endTime = performance.now()
+    ReactGA.timing({
+      category: 'PatternImport',
+      variable: 'read' + importer.label,
+      value: this.endTime - this.startTime // in milliseconds
+    })
+  }
+
+  onFileSelected(event) {
+    let file = event.target.files[0]
+    let reader = new FileReader()
+
+    reader.onload = (event) => {
+      this.startTime = performance.now()
+      var text = reader.result
+
+      let importer
+      if (file.name.toLowerCase().endsWith('.thr')) {
+        importer = new ThetaRhoImporter(file.name, text)
+      } else if (file.name.toLowerCase().endsWith('.gcode') || file.name.toLowerCase().endsWith('.nc')) {
+        importer = new GCodeImporter(file.name, text)
+      }
+
+      importer.import(this.onFileImported.bind(this))
+    }
+
+    reader.readAsText(file)
+  }
+
+  render() {
+    var commentsRender = this.props.comments.map((comment, index) => {
+      return <span key={index}>{comment}<br/></span>
+    })
+
+    return (
+      <div>
+        <Card className="p-3 pb-4">
+          <Accordion className="mb-4">
+            <Card>
+              <Card.Header as={Form.Label} htmlFor="fileUpload" style={{ cursor: "pointer" }}>
+                <h3>Import</h3>
+                Imports a pattern from a .thr, .gcode, or .nc file.
+                <Form.Control
+                    id="fileUpload"
+                    type="file"
+                    accept=".thr,.gcode,.nc"
+                    onChange={this.onFileSelected.bind(this)}
+                    style={{ display: "none" }} />
+              </Card.Header>
+            </Card>
+          </Accordion>
+
+          <CheckboxOption
+            onChange={this.props.onChange}
+            options={this.props.options}
+            optionKey="aspectRatio"
+            key="aspectRatio"
+            index={1}
+            model={this.props} />
+
+          <InputOption
+            onChange={this.props.onChange}
+            options={this.props.options}
+            key="zoom"
+            optionKey="zoom"
+            index={2}
+            model={this.props} />
+
+          { this.props.fileName && <div id="importer-comments" className="mt-4 p-3">
+            Name: {this.props.fileName} <br />
+            Number of points: {this.props.vertices.length } <br />
+            Comments:
+            <div className="ml-3">
+              { commentsRender }
+            </div>
+          </div>}
+        </Card>
+
+        <div className="p-4">
```

Comment:
This is clearly not an "InputOption" or any other kind of option. I would still like to move this stuff into the "draw" tab like we talked about, but we want some of this information to go with it (and some to end up in the about tab).

Does it make sense to have some other kind of "option" that can just hold stuff like this? I'm not sure how the commentsReader stuff will work either. I know I could make something like a `CheckboxOption` that would just be for drawing comments. Is that the right way to go? Or should it be some kind of `StaticOption` that just contains whatever html we want?

Reply:
I feel like we need to accept more than HTML, since these are all react objects.

Reply:
I agree we'll need to deal with this, but I think it can wait until import adds a new shape to the playlist.